pkg/ebpf/c/common/buffer.h

```diff
@@ -82,29 +83,40 @@ statfunc int save_to_submit_buf(args_buffer_t *buf, void *ptr, u32 size, u8 inde
 {
     // Data saved to submit buf: [index][ ... buffer[size] ... ]
 
-    if (size == 0)
+    barrier();
+    // set argument size bounds
+    if (size == 0 || size > MAX_ELEMENT_SIZE || buf->offset >= ARGS_BUF_SIZE)
         return 0;
 
-    barrier();
-    if (buf->offset > ARGS_BUF_SIZE - 1)
+    u32 buffer_index_offset = buf->offset + 1; // buffer offset after writing the index
+    if (buffer_index_offset > ARGS_BUF_SIZE)
         return 0;
 
-    // Save argument index
+    // Save argument index (add 1 to index later)
```

Comment:
Where this increment is done?

Reply:
I think I meant `argnum` here as it should, under ideal conditions, correspond to the index.